text/0517-io-os-reform.md

```diff
@@ -556,6 +556,15 @@ buffers as input. This has multiple benefits:
   results are wanted, one can use `read_to_end` and convert to a
   `String` only at the end.
 
+`read_to_end` and `read_to_string` return a `Result` with either the
+number of bytes read or an error.  If a short read occurs, that call
```

Comment:
In this context, what would you define as a "short read"? If you're asking for the entire contents of the stream then a "short read" implies you already know how many bytes are in the stream in the first place.

Additionally, below you mention a "non-transient" error, but how do we classify errors as transient or not transient?